### PR TITLE
[requirements] Update dependencies & drop python 2 support #39

### LINKED ISSUE #39: [requirements] Update dependencies & drop python 2 support
Python2 is very close to the end of it's lifecycle, it's time we drop the support for Python 2.

To do that,
1. We need to update all the dependencies to python3.
2. Remove code that exists for python 2 support.
3. Remove python2 from README & any documentation.

### PR DESCRIPTION
**Pull request by IAmAMissingColon**
Upgraded dpendencies and removed python2 structures.
**Files Modified**
* .travis.yml
* openwisp_network_topology/apps.py
* openwisp_network_topology/models.py
* openwisp_network_topology/tests/test_admin.py 
* openwisp_network_topology/tests/test_utils.py
* requirements.txt 
* setup.py

Closes #39

### CODE REVIEW

**Comment by IAmAMissingColon** [comment]
There are some errors present. Fixing them :smile: 
And is there's anything to look out for ? :smile:

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 django-netjsongraph>=0.5.0,<0.6.0
-openwisp-users>=0.1.12,<0.2
-openwisp-utils>=0.3.2,<0.4.0
+openwisp-users>=0.1.12,<0.3
+openwisp-utils>=0.3.2,<0.5.0
```
Comment (atb00ker) [inline requirements.txt]: Can you set the minimum to be 0.4 please? :smile:
Testing the rest.

**Comment by atb00ker** [comment]
Travis is failing, can you please check it! :smile:
Also, please drop python 3.5 and update the README accordingly! :smile:

**Comment by atb00ker** [comment]
I also see: `from __future__ import unicode_literals`

File: requirements.txt (hunk 1 above)
Comment (IAmAMissingColon) [reply requirements.txt]: Done :smile:

**Comment by IAmAMissingColon** [comment]
`error: openwisp-utils 0.4.1 is installed but openwisp-utils<0.4,>=0.3 is required by {'django-netjsongraph'}`

File: .travis.yml [hunk 2]
```diff
@@ -2,25 +2,22 @@ language: python
...
   - DJANGO="django>=2.1,<2.2"
   - DJANGO="django>=2.2,<2.3"
-
-matrix:
-  exclude:
-   - python: "2.7"
-     env: DJANGO="django>=2.1,<2.2"
-   - python: "2.7"
-     env: DJANGO="django>=2.2,<2.3"
+  - DJANGO="django>=2.3,<3.1"
```
Comment (nemesisdesign) [inline .travis.yml]: should be `- DJANGO="django>=3.0,<3.1"`

File: .travis.yml [hunk 3]
```diff
@@ -2,25 +2,22 @@ language: python
...
 
 python:
-  - "3.5"
-  - "2.7"
+  - "3.6"
+  - "3.7"
 
 env:
-  - DJANGO="django>=1.11,<1.12"
   - DJANGO="django>=2.1,<2.2"
```
Comment (nemesisdesign) [inline .travis.yml]: remove this line as well

File: .travis.yml (hunk 2 above)
Comment (IAmAMissingColon) [reply .travis.yml]: :+1:

File: .travis.yml (hunk 3 above)
Comment (IAmAMissingColon) [reply .travis.yml]: :+1:

**Comment by nemesisdesign** [comment]
@IAmAMissingColon the build is still failing

**Comment by IAmAMissingColon** [comment]
Almost There

File: openwisp_network_topology/migrations/0001_initial.py [hunk 4]
```diff
@@ -1,16 +1,18 @@
 # -*- coding: utf-8 -*-
 # Generated by Django 1.11.3 on 2017-07-13 13:31
-from __future__ import unicode_literals
-
+ 
 from django.db import migrations, models
+from openwisp_utils.utils import get_random_key
```
Comment (IAmAMissingColon) [inline openwisp_network_topology/migrations/0001_initial.py]: Added since `get_random_key` moved to `opewisp_utils`

File: .travis.yml [hunk 5]
```diff
@@ -2,32 +2,32 @@ language: python
...
   only:
     - master
 
+before_install:
+  - pip install -U pip
+  - pip install -U pip wheel setuptools
+
 install:
   - pip install $DJANGO
+  - pip install -e git+git://github.com/openwisp/django-netjsongraph#egg=django-netjsongraph
```
Comment (TheOneAboveAllTitan) [inline .travis.yml]: I suggest locking it to some specific commit otherwise, the build will probably fail with changes in `django-netjsongraph`.

Comment (nemesisdesign) [reply .travis.yml]: don't worry, I will handle this once I release django-netjsongraph

File: openwisp_network_topology/migrations/0001_initial.py [hunk 6]
```diff
@@ -61,7 +63,7 @@ class Migration(migrations.Migration):
                 ('parser', models.CharField(choices=[('netdiff.OlsrParser', 'OLSRd (txtinfo/jsoninfo)'), ('netdiff.BatmanParser', 'batman-advanced (jsondoc/txtinfo)'), ('netdiff.BmxParser', 'BMX6 (q6m)'), ('netdiff.NetJsonParser', 'NetJSON NetworkGraph'), ('netdiff.CnmlParser', 'CNML 1.0')], help_text='Select topology format', max_length=128, verbose_name='format')),
                 ('strategy', models.CharField(choices=[('fetch', 'FETCH'), ('receive', 'RECEIVE')], db_index=True, default='fetch', max_length=16, verbose_name='strategy')),
                 ('url', models.URLField(blank=True, help_text='Topology data will be fetched from this URL (FETCH strategy)', verbose_name='url')),
-                ('key', models.CharField(blank=True, default=django_netjsongraph.utils.get_random_key, help_text='key needed to update topology from nodes ', max_length=64, verbose_name='key')),
+                ('key', models.CharField(blank=True, default=get_random_key, help_text='key needed to update topology from nodes ', max_length=64, verbose_name='key')),
```
Comment (nemesisdesign) [inline openwisp_network_topology/migrations/0001_initial.py]: in order to fix the build you need to change this line in a similar way to how it was changed in django-netjsongraph, see https://github.com/openwisp/django-netjsongraph/commit/110cb7175526bfe105d5a6a16f58085dcf6a5103#diff-8d08650f6724971ea8f07469294f2f33L27

Comment (IAmAMissingColon) [reply openwisp_network_topology/migrations/0001_initial.py]: :+1:

**Comment by nemesisdesign** [comment]
Thanks! :+1:

**Pull request merged**